src/HeatmapBuilder.php

```diff
@@ -125,10 +130,13 @@ protected function mapToCarbon(array|ArrayAccess|TimeGroupable $item): Carbon
             ? Carbon::parse($item->getTimeGroupableIndicator())
             : Carbon::parse($item[$this->arrayAccessIndex]);
 
+        $date->setTimezone($this->timezone);
```

Comment:
Let's add a condition, if timezone was passed to the function, set the timezone, if not passed, don't set it.

Reply:
time zone can`t be null, i added condition where if $timezone=null set timezone to utc.